### PR TITLE
docs: add features to Current Flagsmith Feature Flags

### PR DESCRIPTION
**Pull request by gnumoreno**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Adding more internal flagsmith feature flags.

## How did you test this code?

Simple doc change. Preview looks fine.

### CODE REVIEW

File: docs/docs/deployment/overview.md [hunk 1]
```diff
@@ -179,31 +179,56 @@ the platform. For example, if you wanted to disable Google OAuth authentication,
...
+| `configure_hide_sensitive_data`       | If the value is true, the hide sensitive data switch will be displayed in the environment settings.                                             | None                                                                                                         |
+| `consistent_hashing_setting`          | Displays the setting to control the `use_identity_composite_key_for_hashing` environment attribute                                              | None                                                                                                         |
+| `dark_mode`                           | Enables Dark Mode in UI See Below                                                                                                               | None                                                                                                         |
+| `demo_feature`                        | Shows a demo feature in the side navigation bar!                                                                                                | green                                                                                                        |
```
Comment (matthewelwell) [inline docs/docs/deployment/overview.md]: This one doesn't need to be included in the docs.

File: docs/docs/deployment/overview.md [hunk 2]
```diff
@@ -179,31 +179,56 @@ the platform. For example, if you wanted to disable Google OAuth authentication,
...
+| `read_only_mode`                      | Determines if org needs to contact sales                                                                                                        | None                                                                                                         |
+| `rotate_api_token`                    | Enables the ability to rotate a user's access token                                                                                             | [See Below](#oauth_google)                                                                                   |
+| `saml`                                | Enables SAML authentication                                                                                                                     | [See](deployment/configuration/authentication/SAML)                                                          |
+| `scaleup_audit`                       | Disables audit log for anyone under scale-up plan                                                                                               | None                                                                                                         |
```
Comment (matthewelwell) [inline docs/docs/deployment/overview.md]: This one doesn't need to be included in the docs.

File: docs/docs/deployment/overview.md [hunk 3]
```diff
@@ -179,31 +179,56 @@ the platform. For example, if you wanted to disable Google OAuth authentication,
...
+| `show_role_management`                | Show role management tab in OrganisationalSettingsPage                                                                                          | None                                                                                                         |
+| `sso_idp`                             | For self hosted, this will automatically redirect to the pre configured IdP                                                                     | None                                                                                                         |
+| `tag_environments`                    | Enables an environment setting to add a UI hint to your environments (e.g. for prod)                                                            | None                                                                                                         |
+| `try_it`                              | Whether to show the try it buttons                                                                                                              | None                                                                                                         |
```
Comment (matthewelwell) [inline docs/docs/deployment/overview.md]: This one doesn't need to be included in the docs.

File: docs/docs/deployment/overview.md [hunk 4]
```diff
@@ -179,31 +179,56 @@ the platform. For example, if you wanted to disable Google OAuth authentication,
...
+| `usage_chart`                         | Organisation Analytics usage chart - requires InfluxDB                                                                                          | None                                                                                                         |
```
Comment (matthewelwell) [inline docs/docs/deployment/overview.md]: This no longer needs InfluxDB. 

```suggestion
| `usage_chart`                         | Organisation Analytics usage chart - requires [additional infrastructure](/deployment/overview#flag-analytics)                                                                                         | None                                                                                                         |
```

File: docs/docs/deployment/overview.md [hunk 5]
```diff
@@ -179,31 +179,56 @@ the platform. For example, if you wanted to disable Google OAuth authentication,
...
+| `is_server_key_only`                  | Adds is_server_key_only setting for flags                                                                                                       | None                                                                                                         |
+| `mailing_list`                        | Determines if mailing list consent is shown on signup                                                                                           | None                                                                                                         |
+| `max_api_calls_alert`                 | If enabled, shows an alert message in the top banner when the organization is over a 70% of its API calls limit                                 | None                                                                                                         |
+| `not_operator`                        | Adds a not operator to segment rules, add an array of sdks to denote client side evaluation support.                                            | None                                                                                                         |
```
Comment (matthewelwell) [inline docs/docs/deployment/overview.md]: This one shouldn't be included in the docs - the BE does not support it.

File: docs/docs/deployment/overview.md [hunk 6]
```diff
@@ -179,31 +179,56 @@ the platform. For example, if you wanted to disable Google OAuth authentication,
...
+|---------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------|
+| `4eyes`                               | Whether to enable the Change Requests feature                                                                                                   | None                                                                                                         |
+| `announcement`                        | Shows an announcement at the top of the app                                                                                                     | None                                                                                                         |
+| `annual_plans`                        | Adds annual plans to pricing                                                                                                                    | None                                                                                                         |
```
Comment (matthewelwell) [inline docs/docs/deployment/overview.md]: This one shouldn't be included in the docs (SaaS only feature)

File: docs/docs/deployment/overview.md [hunk 7]
```diff
@@ -179,31 +179,56 @@ the platform. For example, if you wanted to disable Google OAuth authentication,
...
+| `broken_feature`                      | Exposes a page to crash the Flagsmith app, to demo with Dynatrace observability                                                                 | None                                                                                                         |
```
Comment (matthewelwell) [inline docs/docs/deployment/overview.md]: This one shouldn't be included in the docs (used for demoing only)

File: docs/docs/deployment/overview.md [hunk 8]
```diff
@@ -179,31 +179,56 @@ the platform. For example, if you wanted to disable Google OAuth authentication,
...
+| `demo_feature`                        | Shows a demo feature in the side navigation bar!                                                                                                | green                                                                                                        |
+| `disable_create_org`                  | Turning this on will prevent users from creating any additional organisations                                                                   | None                                                                                                         |
+| `disable_users_as_reviewers`          | If enabled, this flag will hide the Assigned users section in the Change Requests and in the Create Change Request modal in the Features page.  | None                                                                                                         |
+| `edge_identities`                     | Uses the new edge api for identities if the project is setup                                                                                    | None                                                                                                         |
```
Comment (matthewelwell) [inline docs/docs/deployment/overview.md]: This one shouldn't be included in the docs (SaaS only feature)

File: docs/docs/deployment/overview.md [hunk 9]
```diff
@@ -179,31 +179,56 @@ the platform. For example, if you wanted to disable Google OAuth authentication,
...
+| `enable_metadata`                     | If enabled, metadata can be handled                                                                                                             | None                                                                                                         |
+| `feature_name_regex`                  | Enables the project setting to add a regex matcher to validate feature names                                                                    | None                                                                                                         |
+| `feature_versioning`                  | Opt into feature versioning for your environment                                                                                                | None                                                                                                         |
+| `flag_analytics`                      | Flag usage chart - requires additional infrastructure (See here)                                                                                | None                                                                                                         |
```
Comment (matthewelwell) [inline docs/docs/deployment/overview.md]: Looks like we've lost the link on this one? `(See _here_)` should link to (/deployment/overview#flag-analytics)

File: docs/docs/deployment/overview.md [hunk 10]
```diff
@@ -179,31 +179,56 @@ the platform. For example, if you wanted to disable Google OAuth authentication,
...
+| `oauth_google`                        | Google login key                                                                                                                                | [See Below](#oauth_google)                                                                                   |
+| `payments_enabled`                    | Determines whether to show payment UI / seats                                                                                                   | None                                                                                                         |
+| `plan_based_access`                   | Controls rbac and 2f based on plans                                                                                                             | None                                                                                                         |
+| `read_only_mode`                      | Determines if org needs to contact sales                                                                                                        | None                                                                                                         |
```
Comment (matthewelwell) [inline docs/docs/deployment/overview.md]: These shouldn't be included in the docs (SaaS only features)

**Review by matthewelwell — DISMISSED** [review]
I've added a few comments to the list of flags. Some need removing.

**Review by dabeeeenster — APPROVED** [review]

**Pull request merged**